service_api.md

```diff
@@ -0,0 +1,284 @@
+full resource and perform an additional service request.
+
+### Update a specific project
+
+```mermaid
+---
+title: Update Funding Information Example
+---
+classDiagram
+    class AsyncProjectService {
+        Mono~ProjectUpdateResponse~ update(ProjectUpdateRequest request)
+    }
+    <<interface>> AsyncProjectService
+    AsyncProjectService ..> ProjectUpdateRequest
+    AsyncProjectService ..> ProjectUpdateResponse
+
+    class ProjectUpdateRequest {
+        String projectId
+        ProjectUpdateRequestBody requestBody
+        String requestId
+    }
+
+    class ProjectUpdateResponse {
+        String projectId
+        ProjectUpdateResponseBody responseBody
+        String requestId
+    }
+
+    class ProjectUpdateRequestBody {
+    }
+    <<interface>> ProjectUpdateRequestBody
+
+    class ProjectUpdateResponseBody {
+    }
+    <<interface>> ProjectUpdateResponseBody
+
+    class FundingInformation
+    FundingInformation ..|> ProjectUpdateRequestBody
+    FundingInformation ..|> ProjectUpdateResponseBody
+    ProjectUpdateRequest ..> ProjectUpdateRequestBody
+    ProjectUpdateResponse ..> ProjectUpdateResponseBody
+
+
+
+```
+
+### Update project contacts
+
+```mermaid
+---
+title: Update Project Contacts Example
+---
+classDiagram
+    class AsyncProjectService {
+        Mono~ProjectUpdateResponse~ update(ProjectUpdateRequest request)
+    }
+    <<interface>> AsyncProjectService
+    AsyncProjectService ..> ProjectUpdateRequest
+    AsyncProjectService ..> ProjectUpdateResponse
+
+    class ProjectUpdateRequest {
+        String projectId
+        ProjectUpdateRequestBody requestBody
+        String requestId
+    }
+
+    class ProjectUpdateResponse {
+        String projectId
+        ProjectUpdateResponseBody responseBody
+        String requestId
+    }
+
+    class ProjectUpdateRequestBody {
+    }
+    <<interface>> ProjectUpdateRequestBody
+
+    class ProjectUpdateResponseBody {
+    }
+    <<interface>> ProjectUpdateResponseBody
+
+    class ProjectContacts
+    ProjectContacts ..|> ProjectUpdateRequestBody
+    ProjectContacts ..|> ProjectUpdateResponseBody
+    ProjectUpdateRequest ..> ProjectUpdateRequestBody
+    ProjectUpdateResponse ..> ProjectUpdateResponseBody
+```
+
+### Update of a sub-resource with multiplicity
+
+Sometimes a resource contains more than one sub-resource of the same type. This happens, when there
+is a one-to-many relationship between the resource and its sub-resource. For example an experiment can
+contain one or more experimental groups.
+
+To enable clients to update a certain sub-resource, a combination of the parent resource's ID and the local 
+ID of the sub-resource is required.
+
+For example to update an `Experimental Group` of an `Experiment`, the API shall provide a method like `update(ExperimentalGroupUpdateRequest request)`. The object request needs to provide
+the required IDs to ensure that the service can perform the task.
```

Comment:
When creating/updating multiple sub-resources, how do we deal with rollback strategies? What do we do in case of failures? When we return only the changed information we do not have the possibility to rollback single executions? Do we require each request in the update method to be `all-or-nothing`? So will a desired update of multiple groups lead to a request body with information about multiple groups or do we want to have multiple calls to update with multiple request bodies with a single group each?

Reply:
If there is a use case, where a user wants to submit an update of more than one sub-resource, then we need to extend the API for it (e.g. a list of `ExperimentalGroup`. 


Each request is transactional for the resource.